test_libs/pyspec/eth2spec/test/epoch_processing/test_process_registry_updates.py

```diff
@@ -19,15 +20,12 @@ def test_activation(state):
     state.validator_registry[index].effective_balance = spec.MAX_EFFECTIVE_BALANCE
     assert not is_active_validator(state.validator_registry[index], get_current_epoch(state))
 
-    yield 'pre', state
-
-    blocks = []
     for _ in range(spec.ACTIVATION_EXIT_DELAY + 1):
-        block = next_epoch(state)
-        blocks.append(block)
+        next_epoch(state)
+
+    yield 'pre', state
 
-    # provide extra type hinting here, since it is wrapped in a list.
-    yield 'blocks', blocks, [spec.BeaconBlock]
+    process_registry_updates(state)
```

Comment:
It seems a little bit artificial to run `process_registry_updates` on a state that has not yet undergone the functions in the slot processing prior to it. (`process_slot` and the three preceding  functions in `process_epoch`).

Without getting the state to this mid-point in the epoch transition, many things we expect to happen might not. 

I would consider adding a `prepare_state_for_registry_update` helper that is called before yielding the `pre` state. This function would step up to the next epoch transition slot and trigger the preceding function calls.

Reply:
It's more similar to crosslinks now (see 73f0f74), with a `run_process_registry_updates` function